### PR TITLE
180 kill enemy when they hit a projectile

### PR DESCRIPTION
**Pull request by LucasTesnier**
Fix some bugs relative to the Death system :
- Implement a correct transit for Death component (No suppression before sending)
- Add a verification inside the Collide systems before removing a lifePoint
- Delete the new enemy creation at the last enemy death

Implement a new feature :
- When your player died, the game room is passed in ENDED state.

### CODE REVIEW

**Review by 31Nathan — DISMISSED** [review]
GG

**Review by osvegn — CHANGES_REQUESTED** [review]
Just a question for the documentation

File: Client/ClientRoom.hpp [hunk 1]
```diff
@@ -53,6 +53,9 @@ namespace client_data
         /// @brief Init some entities only created by the client.
         void _initEntities();
 
+        /// @brief Check if the player still alive and update the gamestate if his dead
```
[the remark this answers is not in the document]
Comment (LucasTesnier) [reply Client/ClientRoom.hpp:56]: I thinks his his a beautiful use of the english language

File: Client/ClientRoom.hpp [hunk 2]
```diff
@@ -53,6 +53,9 @@ namespace client_data
         /// @brief Init some entities only created by the client.
         void _initEntities();
 
+        /// @brief Check if the player still alive and update the gamestate if is dead
+        void _holdGameOver(void);
```
Comment (ZQUEMA) [inline Client/ClientRoom.hpp:57]: No need to specify the parameter `void`

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
One minor change for now

File: Client/ClientRoom.hpp (hunk 2 above)
Comment (LucasTesnier) [reply Client/ClientRoom.hpp:57]: I am not okay with that, i used the C convention

Comment (osvegn) [reply Client/ClientRoom.hpp:57]: But we're using cpp, you should remove it

**Review by ZQUEMA — DISMISSED** [review]

**Review by osvegn — DISMISSED** [review]
Look good

File: libs/R-TypeLogic/Server/Systems/CollidableSystem.cpp [hunk 3]
```diff
@@ -51,15 +51,19 @@ void Collide::collide(
             if (isSameHeight(fstPos, sndPos, fstSize, sndSize) && isSameWidth(fstPos, sndPos, fstSize, sndSize)) {
                 Damage &fstDamage = fstEntity->getComponent<Damage>();
                 Damage &sndDamage = sndEntity->getComponent<Damage>();
+                Life &fstLife = fstEntity->getComponent<Life>();
```
Comment (31Nathan) [inline libs/R-TypeLogic/Server/Systems/CollidableSystem.cpp:54]: If obstacle, there is no life so there may be a throw uncatch

**Review by 31Nathan — CHANGES_REQUESTED** [review]

**Review by 31Nathan — APPROVED** [review]
LGTM

**Review by osvegn — APPROVED** [review]
Look good

**Pull request merged**